detd/manager.py

```diff
@@ -86,7 +86,7 @@ def add_talker(self, config):
         with self.lock:
 
             if not config.interface.name in self.talker_manager:
-                interface_manager = InterfaceManager(config.interface, config.options)
+                interface_manager = InterfaceManager(config.interface, config.hints)
```

Comment:
Just pass to the constructor "config". Making the details visible at this point probably does not help much.